xapian-core/tests/harness/backendmanager_remotetcp.cc

```diff
@@ -364,41 +374,65 @@ BackendManagerRemoteTcp::get_writable_database(const string & name,
 					       const string & file)
 {
     string args = get_writable_database_args(name, file);
-    int port = launch_xapian_tcpsrv(args);
-    return Xapian::Remote::open_writable(LOCALHOST, port);
+	auto pp = launch_xapian_tcpsrv(args);
+	auto db = Xapian::Remote::open_writable(LOCALHOST, pp.port);
+	uuid_to_pididx[db.get_uuid()] = pp.pididx;
+	return db;
 }
 
 Xapian::Database
 BackendManagerRemoteTcp::get_remote_database(const vector<string> & files,
 					     unsigned int timeout)
 {
     string args = get_remote_database_args(files, timeout);
-    int port = launch_xapian_tcpsrv(args);
-    return Xapian::Remote::open(LOCALHOST, port);
+    auto pp = launch_xapian_tcpsrv(args);
+    return Xapian::Remote::open(LOCALHOST, pp.port);
```

Comment:
Here you don't record the mapping needed to kill the server, nor below.

Rather than add the same handling in 3 places, it would be better for `launch_xapian_tcpsrv()` to take care of updating that mapping, and then it can just return the port number as it currently does.